util/syn_yosys.sh

```diff
@@ -10,64 +10,120 @@
 # The following tools are required:
 #  - sv2v: SystemVerilog-to-Verilog converter from github.com/zachjs/sv2v
 #  - yosys: synthesis tool from github.com/YosysHQ/yosys
+#  - Cadence Conformal
 #
 # Usage:
-#   syn_yosys.sh 2>&1 | tee syn.std
+#   ./syn_yosys.sh 2>&1 | tee syn.std
 
 #-------------------------------------------------------------------------
 # use fusesoc to generate files and file list
 #-------------------------------------------------------------------------
-\rm -Rf build
-fusesoc --cores-root .. run --target=syn --setup lowrisc:systems:top_earlgrey
+\rm -Rf build syn_out
+fusesoc --cores-root .. run --target=syn \
+  --setup lowrisc:systems:top_earlgrey > /dev/null 2>&1
 
 # copy all files into directory "syn_out"
-\rm -Rf syn_out
 mkdir syn_out
 cp \
-  build/lowrisc_systems_top_earlgrey_0.1/src/*/*.sv \
-  build/lowrisc_systems_top_earlgrey_0.1/src/*/*/*.sv \
-  build/lowrisc_systems_top_earlgrey_0.1/src/*/*/*/*.sv \
+  build/*/src/*/*.sv* \
+  build/*/src/*/*/*.sv* \
+  build/*/src/*/*/*/*.sv* \
   syn_out
 cd syn_out
 
-# not synthesizable
-\rm prim_pad_wrapper.sv
-\rm prim_generic_pad_wrapper.sv
+# copy file list, remove incdir lines, flatten pathnames
+grep -v incdir ../build/*/*/*.scr | sed 's!.*/!!' > flist_gold
 
-# not used for synthesis
-\rm pins_if.sv
-\rm tlul_assert.sv
-\rm tlul_assert_multiple.sv
-
-# match filename to module name
-mv ibex_register_file{_ff,}.sv
-
-# prim_util_memload.sv is only meant to be included within a module
-mv prim_util_memload.sv{,h}
-sed -i.delete.bak -e "s/prim_util_memload\.sv/prim_util_memload.svh/" *.sv
-rm *.delete.bak  # some sed implementations require backup extensions
+# generate revised flist by replacing '.sv' by '.v' and removing packages
+sed 's/.sv/.v/g' flist_gold | grep -v "_pkg.v" > flist_rev
 
 #-------------------------------------------------------------------------
 # convert all RTL files to Verilog
 #-------------------------------------------------------------------------
-printf "\n\nsv2v:\n"
-sv2v -DSYNTHESIS *.sv > combined.v
-# split files up
-modules=`cat combined.v | grep "^module" | sed -e "s/^module //" | sed -e "s/ (//"`
-echo "$modules" > modules.txt  # for debugging
-for module in $modules; do
-  sed -n "/^module $module /,/^endmodule/p" < combined.v > $module.v
+
+# hack SystemVerilog files to avoid SV2V and/or LEC issues
+# TODO: eventually remove these hacks
+
+# 1) Setting ByteAccess = 0 for tlul_adapter_sram.sv doesn't work. This
+# hack changes the functionality so needs fixing
+sed -i 's/ByteAccess *(0)/ByteAccess(1)/g' *.sv
```

Comment:
Is this referring to an issue in sv2v?

Reply:
I wasn't able to figure it out yet, it could also be an LEC issue. LEC only passes when I set ByteAccess to 1.